pkg/webservice/handlers.go

```diff
@@ -339,3 +342,55 @@ func getClusterConfig(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
+
+var mutex sync.Mutex
+func updateConfig(w http.ResponseWriter, r *http.Request) {
+	mutex.Lock()
+	defer mutex.Unlock()
+	writeHeaders(w)
+	requestBytes, err := ioutil.ReadAll(r.Body)
+
+	if err == nil {
+		// validation is already called when loading the config
+		schedulerConf, err2 := configs.LoadSchedulerConfigFromByteArray(requestBytes)
+		if err2 != nil {
+			buildUpdateResponse(false, err2.Error(), w)
+			return
+		}
+		oldConf, err3 := saveConfigmap(string(requestBytes))
+		if err3 != nil {
+			buildUpdateResponse(false, err3.Error(), w)
+			return
+		}
+		err4 := gClusterInfo.UpdateSchedulerConfig(schedulerConf)
+		if err4 != nil {
+			// revert configmap changes
+			saveConfigmap(oldConf)
+			buildUpdateResponse(false, err4.Error(), w)
+			return
+		}
+		buildUpdateResponse(true, "", w)
+	}
+}
+
+func buildUpdateResponse(success bool, reason string, w http.ResponseWriter) {
+	var result dao.UpdateConfResponse
+	result.Success = success
+	result.Reason = reason
+	if err := json.NewEncoder(w).Encode(result); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+func saveConfigmap(conf string) (string, error) {
```

Comment:
Rename `saveConfigmap` to `updateConfiguration`

Reply:
Done